File: Serene/Serene.Script/Imports/ServiceContracts/BasicSamples.BasicSamplesService.cs
namespace Serene.BasicSamples
{
    using jQueryApi;
    using Serenity;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [Imported, PreserveMemberCase]
    public partial class BasicSamplesService
    {
        [InlineConstant] public const string BaseUrl = "BasicSamples/BasicSamples";
    
        [InlineCode("Q.serviceRequest('BasicSamples/BasicSamples/OrdersByShipper', {request}, {onSuccess}, {options})")]
        public static jQueryXmlHttpRequest OrdersByShipper(OrdersByShipperRequest request, Action<OrdersByShipperResponse> onSuccess, ServiceCallOptions options = null)
        {
            return null;
        }
    
        [Imported, PreserveMemberCase]
        public static class Methods
        {
            [InlineConstant] public const string OrdersByShipper = "BasicSamples/BasicSamples/OrdersByShipper";
        }
    }
    
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/BasicSamples.BasicSamplesService.ts
namespace Serene.BasicSamples {
    export namespace BasicSamplesService {
        export const baseUrl = 'BasicSamples/BasicSamples';

        export declare function OrdersByShipper(request: OrdersByShipperRequest, onSuccess?: (response: OrdersByShipperResponse) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;
        export declare function OrderBulkAction(request: OrderBulkActionRequest, onSuccess?: (response: Serenity.ServiceResponse) => void, opt?: Serenity.ServiceOptions<any>): JQueryXHR;

        export namespace Methods {
            export declare const OrdersByShipper: string;
            export declare const OrderBulkAction: string;
        }

        ['OrdersByShipper', 'OrderBulkAction'].forEach(x => {
            (<any>BasicSamplesService)[x] = function (r, s, o) { return Q.serviceRequest(baseUrl + '/' + x, r, s, o); };
            (<any>Methods)[x] = baseUrl + '/' + x;
        });
    }
}

